diff --git a/gfx/drivers_context/drm_go2_ctx.c b/gfx/drivers_context/drm_go2_ctx.c
index b8d1b5d598..efef06aab2 100644
--- a/gfx/drivers_context/drm_go2_ctx.c
+++ b/gfx/drivers_context/drm_go2_ctx.c
@@ -337,7 +337,7 @@ static void gfx_ctx_go2_drm_swap_buffers(void *data)
          surf,
          src_x, src_y, src_w, src_h,
          out_x, out_y, out_w, out_h,
-         GO2_ROTATION_DEGREES_0, 2);
+         GO2_ROTATION_DEGREES_180, 2);
    go2_context_surface_unlock(drm->context, surf);
 #endif
 }
