diff --git a/src/globals.cpp b/src/globals.cpp
index 0d81cfa..c17e465 100755
--- a/src/globals.cpp
+++ b/src/globals.cpp
@@ -381,6 +381,15 @@ bool isRG353M()
 {
     return checkDeviceName((char *)"RG353M"); 
 }
+bool isU8() {
+    return checkDeviceName((char *)"U8"); 
+}
+bool isA10miniv2() {
+    return checkDeviceName((char *)"A10miniv2"); 
+}
+bool isXF28() {
+    return checkDeviceName((char *)"XF28"); 
+}
 bool hasDeviceRotatedScreen(){
    return  isRG351V() || isRG351MP() || isRG503() || isRG353V() || isRG353M();
 }
diff --git a/src/globals.h b/src/globals.h
index 5ed5ff6..c076c4d 100755
--- a/src/globals.h
+++ b/src/globals.h
@@ -195,6 +195,9 @@ bool isRG552();
 bool isRG503();
 bool isRG353V();
 bool isRG353M();
+bool isU8();
+bool isXF28();
+bool isA10miniv2();
 bool hasDeviceRotatedScreen();
 bool wideScreenNotRotated();
 void resetDeviceName();
diff --git a/src/main.cpp b/src/main.cpp
index 35bbcad..53bb043 100755
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -2151,7 +2151,7 @@ int main(int argc, char *argv[])
     // but the parameter sesetnd via command line wins so if that is true we leave it true
     if (!gpio_joypad)
     {
-        if (isRG351MP() || isRG552() )
+        if (isRG351MP() || isRG552() || isU8() || isXF28() || isA10miniv2() )
         {
             gpio_joypad = true;
         }
diff --git a/src/video-helper.cpp b/src/video-helper.cpp
index 7d78975..1ad482d 100755
--- a/src/video-helper.cpp
+++ b/src/video-helper.cpp
@@ -137,7 +137,22 @@ go2_rotation getBlitRotation()
 
         if (!isTate() && tateState != REVERSED)
         {
-            return (hasDeviceRotatedScreen()) ? GO2_ROTATION_DEGREES_0 : GO2_ROTATION_DEGREES_270;
+            if (hasDeviceRotatedScreen()) 
+            {
+                return GO2_ROTATION_DEGREES_0;
+            }
+            else if (isA10miniv2())
+            {
+                return GO2_ROTATION_DEGREES_180;
+            }
+            else if (isXF28())
+            {
+                return GO2_ROTATION_DEGREES_90;
+            }
+            else
+            {
+                return GO2_ROTATION_DEGREES_270;
+            }
         }
         if (tateState == REVERSED)
         {
@@ -172,7 +187,22 @@ go2_rotation getRotation()
     { // landscape
         if (!isTate() && tateState != REVERSED)
         {
-            return (hasDeviceRotatedScreen()) ? GO2_ROTATION_DEGREES_0 : GO2_ROTATION_DEGREES_270;
+            if (hasDeviceRotatedScreen())
+            {
+                return GO2_ROTATION_DEGREES_0;
+            }
+            else if (isA10miniv2())
+            {
+                return GO2_ROTATION_DEGREES_180;
+            }
+            else if (isXF28())
+            {
+                return GO2_ROTATION_DEGREES_90;
+            }
+            else
+            {
+                return GO2_ROTATION_DEGREES_270;
+            }
         }
         if (tateState == REVERSED)
         {
